### ISSUE TITLE
Proposal:Enable developers to create controls that support selection

### ISSUE THREAD

**Issue by mdmozibur** [microsoft/microsoft-ui-xaml#205]
# Proposal: Make Selector class constructor public 

## Summary
`Selector` - the base class for `ComboBox` and other `ItemsControl` with single selection mode, has a contructor that's not public. As a result, it's being difficult or Impossible to create an `ItemsControl` that has only Single selection mode like `ComboBox`.

**Comment by jevansaks** [comment]
Totally agree! We should do this.

**Comment by micahl** [comment]
@MuziburRahman, thanks for the proposal!  I've been out of the office since just a few days before this was opened so apologies for the slow response.  Is there a specific control you'd like to build where you'd like to start from ItemsControl versus a more derived type that already supports selection behaviors such as ListBox?

**Comment by MuziburRahman** [comment]
@micahl Yes, I was trying to build a custom control that is almost like a `ToggleSplitButton`, except for the fact that it will use ItemsControl as the base class. I was almost done with it, but then had one little annoying problem (please have a look at [this](https://stackoverflow.com/questions/53571737/itemspresenter-in-a-popup-of-a-templated-itemscontrol-shows-only-1-item-on-first)), which I believe, will not arise if I could use `Selector` as the base class.

**Comment by micahl** [comment]
Ah, I see.  I don't think Selector will necessarily solve the issue you're hitting.  Instead of deriving your SplitToggleButton from ListViewBase try deriving from ListView.  And if you anticipate that the number of items may be quite large then be sure to explicitly set a height on the PopupBorder to constrain the layout of its children such that virtualization can kick in.

**Comment by MuziburRahman** [comment]
@micahl Thanks. But it was a single selection control. `ListViewBase` is a multi-selection control. `Selector` is the base class for single selection controls. Apart from the problem I described above, if I want to create a single selection control, there's no way I can do that now because of the inability to derive from `Selector`

**Comment by micahl** [comment]
The alternative to deriving from a specific control (an "is-a" approach) is to use composition (a "has-a" approach).  With the latter approach you wrap the control(s) you need and only expose the properties/method/events you want.  Although there's some extra wire up it has the advantage of better encapsulation.  You're not automatically inheriting and exposing a number of things from ItemsControl that you may not care to deal with and that won't automatically be handled for you.  Is your goal to build a control that could be part of a control library for others to consume, or primarily for consumption within your own app?

**Comment by MuziburRahman** [comment]
@micahl both, to be honest.  Primarily for my own app. But I had a plan to make that public if I could build it.

**Comment by micahl** [comment]
Ok, then I definitely think a has-a approach is worth considering.  I'll look at getting an example that combines this need with #217.  It might not be "short" yet, but hopefully it is understandable and something we can work on making shorter over time.
